lib/pb/serializer.rb

```diff
@@ -102,22 +102,24 @@ def parse_field_mask(field_mask)
         end
       end
 
-      # @param input [Google::Protobuf::FieldMask, Symbol, Array<(Symbol,Hash)>, Hash{Symbol=>(Array,Symbol,Hash)}]
-      # @return [Hash{Symbol=>(Array,Hash)}]
+      # @param input [Google::Protobuf::FieldMask, Symbol, Array<(Symbol,Hash)>, Hash{Symbol=>(Array,Symbol,Hash,Proc)}, Proc]
+      # @return [Hash{Symbol=>(Array,Hash,Proc)}]
       def normalize_mask(input)
         if input.kind_of?(Google::Protobuf::FieldMask)
           input = parse_field_mask(input)
         end
 
-        normalized = {}
-
+        input = input.call if input.kind_of?(Proc)
```

Comment:
Alternatively you may check `responds_to?(:call)` like a duck.

Reply:
Hmm... `respond_to?(:call)` seems to be a little too general because I intended `Proc` for just intermediate representation.